craft_platforms/charm/__init__.py

```diff
@@ -0,0 +1,24 @@
+# This file is part of craft-platforms.
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Charm-specific build information for craft-platforms."""
+
+from ._build import DEFAULT_ARCHITECTURES, get_platforms_charm_build_plan
+
+
+__all__ = [
+    "DEFAULT_ARCHITECTURES",
+    "get_platforms_charm_build_plan",
```

Comment:
this can probably be called `get_build_plan` platforms is a parameter and charm is not needed as it is namespaced

Without scoped imports
```python
import craft_plaftorms
plan = craft_platforms.charm.get_platforms_charm_build_plan("ubuntu@24.04", platforms)
```

Versus

```python
import craft_plaftorms
plan = craft_platforms.charm.get_build_plan("ubuntu@24.04", platforms)
```

Reply:
This is to distinguish it from bundle build plans and `bases` charm build plans. Once I've implemented those, I'll make a dispatcher function that handles it called `get_build_plan`